tutorials/3-intro-to-brian-simulations.ipynb

```diff
@@ -0,0 +1,661 @@
+    "axhline(El/mV, ls='-', c='lightgray', lw=3)\n",
+    "plot(statemon.t/ms, statemon.v.T/mV, '-')\n",
+    "xlabel('Time (ms)')\n",
+    "ylabel('v (mV)');"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ahh, that's more like it!\n",
+    "\n",
+    "## Adding input\n",
+    "\n",
+    "Now let's think about a neuron being driven by a sinusoidal input. Let's go back to a leaky integrate-and-fire to simplify the equations a bit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start_scope()\n",
+    "A = 2.5\n",
+    "f = 10*Hz\n",
+    "tau = 5*ms\n",
+    "eqs = '''\n",
+    "dv/dt = (I-v)/tau : 1\n",
+    "I = A*sin(2*pi*f*t) : 1\n",
+    "'''\n",
+    "G = NeuronGroup(1, eqs, threshold='v>1', reset='v=0', method='euler')\n",
+    "M = StateMonitor(G, variables=True, record=True)\n",
+    "run(200*ms)\n",
+    "plot(M.t/ms, M.v[0], label='v')\n",
+    "plot(M.t/ms, M.I[0], label='I')\n",
+    "xlabel('Time (ms)')\n",
+    "ylabel('v')\n",
+    "legend(loc='best');"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So far, so good and the sort of thing we saw in the first tutorial. Now, what if that input current were something we had recorded and saved in a file? In that case, we can use `TimedArray`. Let's start by reproducing the picture above but using `TimedArray`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start_scope()\n",
+    "A = 2.5\n",
+    "f = 10*Hz\n",
+    "tau = 5*ms\n",
+    "# Create a TimedArray and set the equations to use it\n",
+    "t_recorded = arange(int(200*ms/defaultclock.dt))*defaultclock.dt\n",
+    "I_recorded = TimedArray(A*sin(2*pi*f*t_recorded), dt=defaultclock.dt)\n",
+    "eqs = '''\n",
+    "dv/dt = (I-v)/tau : 1\n",
+    "I = I_recorded(t) : 1\n",
+    "'''\n",
+    "G = NeuronGroup(1, eqs, threshold='v>1', reset='v=0', method='linear')\n",
+    "M = StateMonitor(G, variables=True, record=True)\n",
+    "run(200*ms)\n",
+    "plot(M.t/ms, M.v[0], label='v')\n",
+    "plot(M.t/ms, M.I[0], label='I')\n",
+    "xlabel('Time (ms)')\n",
+    "ylabel('v')\n",
+    "legend(loc='best');"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that for the example where we put the ``sin`` function directly in the equations, we had to use the ``method='euler'`` argument because the linear integrator wouldn't work here (try it!). However, ``TimedArray`` is considered to be constant over a time step and so the linear integrator can be used. note that this means you won't get the same behaviour from these two methods for two reasons. Firstly, the numerical integration methods ``linear`` and ``euler`` give slightly different results. Secondly, ``sin`` is not constant over a timestep whereas ``TimedArray`` is.\n",
```

Comment:
> ``TimedArray`` is considered to be constant over a time step

This is slightly wrong (simplifying): a `TimedArray` with a `dt` smaller than the simulation timestep would not be considered constant over a time step.

"note" --> "Note"